src/crypto.js

```diff
@@ -14,16 +14,26 @@
  * limitations under the License.
  */
 
-import {getElementService} from './element-service';
+import {getExistingServiceForWindow} from './service';
+
+/**
+ * An object holding the public key and its hash.
+ *
+ * @typedef {{
+ *   serviceName: string,
+ *   hash: Uint8Array,
+ *   cryptoKey: !webCrypto.CryptoKey
+ * }}
+ */
+export let PublicKeyInfoDef;
 
 /**
  * @param {!Window} window
- * @return {!Promise<!../extensions/amp-analytics/0.1/crypto-impl.Crypto>}
+ * @return {!./service/crypto-impl.Crypto}
  */
 export function cryptoFor(window) {
-  return (/** @type {!Promise<
-      !../extensions/amp-analytics/0.1/crypto-impl.Crypto>} */ (
-      getElementService(window, 'crypto', 'amp-analytics')));
+  return (/** @type {!./service/crypto-impl.Crypto} */ (
+      getExistingServiceForWindow(window, 'crypto')));
```

Comment:
Right -- so I've been unclear why the old API returned a promise and the new one doesn't.  I assume that you're now doing an eager initialization, while it was formerly doing a lazy one?  Is that a problem?  Like, are you incurring some significant initialization cost on pages that don't need crypto services or something?